shared/lsp/registration.ts

```diff
@@ -363,203 +361,106 @@ export async function register({
         fn: (connection: LSPConnection) => Promise<R>
     ) => Promise<R>
 
-    if (supportsWorkspaceFolders) {
-        const connection = await connect({
-            clientRootUri: null,
+    // Supports only one workspace root
+    // TODO this should store a refcount to avoid closing connections other consumers have a reference to
```

Comment:
Makes sense, but _are_ there other consumers? Doesn't our LS run inside the container network where it won't be reachable?

Reply:
This isn't related to the PR. This also runs in the browser so it's only a single user. I'm not sure how useful it would be to do this, but xrefs providers can temporarily hold a connection that may be closed here.

Reply:
> xrefs providers can temporarily hold a connection that may be closed here.

That is indeed the issue.